### DISCUSSION TITLE
State of the project 2026

### DISCUSSION THREAD

**Discussion by katajakasa** [omf2097/openomf#413] [Announcements]
Updated July 8. 2026

So where are we now ?
---------------------------

We are now (more or less) feature complete! There are still small things here and there that needs to be added, but otherwise all the pieces are in place. Now begins the bug hunting season! We will hopefully be also making more releases with much smaller intervals from here on out.

So what next ?
-----------------

These are the things we are working on, in no particular order:

- Gameplay correctness fixes
- Modding improvements for large palette support
- Advanced menu items
- Fire + Ice
- Figure out more animation tags (https://github.com/omf2097/openomf/projects/3) -- this is an ongoing effort, and is mostly discussed on our discord server. We are utilizing Ghidra to reverse-engineer the old project to see what it does. This is slow and very, very, VERY tedious tediousness.
- Overall bugfixing for everything

**Comment by Santorayo3** [comment]
Just wanted to state my support here. I love this project and really looking forward to the tournament mode. Also I'd love to see a port for the OpenDingux OS to play it on my Anbernic.

**Reply by katajakasa** [reply]
Oof, unfortunately we can't really do the ports -- we'd need to have the actual devices, and then some time to work on them... and we are already short on time just getting the game working on common hardware ;)

Still, if people want to put time to it and offer pull requests, we'll probably merge them. As long as they don't make development harder or break other platforms ;)

**Reply by Santorayo3** [reply]
Ok, thanks for the info. :)

**Comment by Santorayo3** [comment]
Also this video might be helpful to figure out what the stats in the original game do:
https://www.youtube.com/watch?v=HVtDAr7dcjM

**Reply by katajakasa** [reply]
I like the comparisons on the video, it might be useful when testing out changes.

What we really REALLY need on the stats front now however are the precise formulas and numbers. For example, how do we apply agility numbers to the base numbers (and what are those base numbers anyways ?), what are the modifiers, etc.

**Comment by johnoestmannmusic** [comment]
Love what you guys are doing! Can't wait to get back into it after a couple decades - and now running smoothly on my Linux laptop ❤️

**Comment by aknight2015** [comment]
This game remains my favorite fighting game of all time. I truly hope development of this port isn't dead.

**Reply by katajakasa** [reply]
Nope, just very slow sometimes :)

**Reply by katajakasa** [reply]
Btw, note that even though we have not made any official releases for a while, we have a development builds from latest source code here https://github.com/omf2097/openomf/releases/tag/latest .

**Reply by aknight2015** [reply]
Awesome. I wasn't sure as a lot of time the "newest" build is just automatically generated using old code. I am so thrilled that this game is still in development.

**Reply by aknight2015** [reply]
> Nope, just very slow sometimes :)

Yeah, I feel that.

**Comment by johnoestmannmusic** [comment]
Just as an FYI, I briefly discussed / linked to this project as part of the entry on One Must Fall: 2097 in the latest issue of [Freeware Creator Magazine](https://johnoestmannmusic.com/freeware-creator-magazine-oct-nov-2024/)

**Reply by aknight2015** [reply]
I didn't even know this magazine existed. Thank you for the link.

**Comment by 8devil3** [comment]
Hi there. 
I'm a big OMF fan, I played it when I was a kid on my 80486 DX100!
Now I'm a dev but only Laravel, not C.

So I'm asking this: is it possible to add mid air combos (juggles) and wall bounce combos to the game (similar to Tekken)?

Do you think it's feasible? 
If yes, do you plan to add them?

**Reply by Vagabond** [reply]
juggles were in the original (called hyper mode), and we do implement them.

**Reply by LegendaryBlueShirt** [reply]
No not hyper mode!  That's enabling certain specials to be performed in the air.  You mean rehit mode!

**Reply by Vagabond** [reply]
my shame is immeasurable
